### DISCUSSION TITLE
When Viewing PR, Show "[username] has started a review" to all users, not just [username]

### DISCUSSION THREAD

**Discussion by derek-pavao** [community/community#4427] [Pull Requests]
I noticed recently that when you start a review on another user's pull request, the reviewer will see a mention that they have started a review. 
Example:
![BPL-1976__SRP_Signup_by_peteigel_·_Pull_Request__436_·_isubscribed_chief-wiggum](https://user-images.githubusercontent.com/1998134/123968137-f8528480-d984-11eb-81e6-4a67d8cd4f22.png)

Currently this is only viewable by the reviewer and not other members of the team. I think this would be useful if everyone with access to the PR could see that a review has been started. 

There are two main use cases I can see:

The first use case is as a reviewer. This could help to reduce duplication of effort. If you come to review a PR and see that other team members are already actively reviewing that PR, you could move on to another PR that maybe no one has looked at yet. 

The second use case is as the author of the PR. If I have an open PR that has been approved. If I could see that another team member has started a review, but not finished it yet, I could be courteous to them and reach out to ask if they plan to finish it or should I just merge. The other team member could have gotten sidetracked and has valuable feedback tucked away in a pending review that currently only the reviewer knows exists.

Thanks for any consideration you give this request.

**Comment by parkerault** [comment]
This is driving me absolutely crazy and I've been bitten by this so many times. I often spend an hour on a review, and nobody knows that I have started one. The worst problem is that it's easy to forget to submit the review, then none of your comments are visible to anyone else.

I would very much like to have a timeline event that says "some-user has started a review" with a button or checkbox for "show pending comments". This simple change would prevent a lot of friction in my team's current code review process.

Edit: sorry, I don't know how to submit a comment instead of an answer.

**Comment by Javran** [comment]
Running into this recently (and I'm pretty sure this won't be the last time) - I commented something and didn't even realized that I have to submit it to make it visible to others. This "Pending" badge is not helping - what is pending? I thought it means pending action for the review, only later that I discovered that it means my comments are not yet submitted.

And no. If I have to read through your PR review docs to know I have to submit review before it's visible to others, the UI leaves much to be desired.

**Comment by andreieftimie** [comment]
This would be a great feature. I've run into the same issues discussed in the the main request and the additional comment as both PR initiator and as reviewer.

**Comment by hbatalhaStch** [comment]
I just ran into this issue, it was my first time reviewing a code and didn't even notice the `Review changes` button in top-right corner. It should be more intuitive, like a message/reminder saying I still have to submit the review

**Comment by axilis-marko** [comment]
This is a big one for me. Has anyone looked into this feature request?

**Comment by alfrol** [comment]
That would be a very useful information. There were multiple times where I was reviewing a PR and the author merged it because someone else already approved it. Our team has an agreement where we post a message to Slack with a request to review the PR and whoever starts reviewing marks the message with a specific emoji. This approach has many drawbacks, someone might forget to mark the message or just ignores it because it's tedious. Having that information displayed in the PR timeline would be so much better.

**Comment by nvaytet** [comment]
We tend to just 'Assign ourselves' to the PR when we start a review, but I agree that it is easy to forget doing so, and having github automatically display this would be great!

That said, I think that it should show only a small timeline item that states "someone has started a review", and **not** show the pending comments, as it does when it is displaying your own unfinished review.

**Reply by axilis-marko** [reply]
I agree with this. Pending comments might also be notes for the reviewer themself to check something, and it can be expanded or deleted based on further research.

As for the assignment, I think a lot of people add reviewers when they open a PR, not the other way around.

**Reply by Lorenzo45** [reply]
This is a good workaround!

I still like OP's original suggestion of making "[username] has started a review" visible to all users (not the pending comments though) so it's automatic and more explicit

It'd also be nice if there was a status for "Reviewers" for an in progress review, maybe a "..." icon

**Comment by vlfig** [comment]
We currently use `<user>-reviewing` labels to indicate we're at it, but that's extra work and sometimes we forget.

Showing in the timeline:
- ` ꙩ <user> started a review <time> ago.`, followed by either 
- ` ꙩ <user> abandoned their review <time> ago.`, or
- ` ꙩ <user> reviewed <time> ago.`

… would be a much better user experience.

**Reply by AstralBat64** [reply]
generally, year 0
we should check all users sometimes to keep the time correct

**Comment by AstralBat64** [comment]
1:00
7:00
0:59
1:01
6:59
